### PR TITLE
Add fuzzing option to abort on internal errors

### PR DESCRIPTION
**Pull request by axic**
Part of #376.

### CODE REVIEW

**Comment by axic** [comment]
This is for afl.

File: README.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ These are to be used via EVMC `set_option`:
 - `metering=true` will enable metering of bytecode at deployment using the [Sentinel system contract] (set to `false` by default)
 - `evm1mode=<evm1mode>` will select how EVM1 bytecode is handled
 - `sys:<alias/address>=file.wasm` will override the code executing at the specified address with code loaded from a filepath at runtime. This option supports aliases for system contracts as well, such that `sys:sentinel=file.wasm` and `sys:evm2wasm=file.wasm` are both valid. **This option is intended for debugging purposes.**
+- `fuzzing=true` will enable fuzzing mode, which aborts on an internal error (as opposed to transmitting the error code to the client). **This option is solely intended for fuzzing purposes.**
```
Comment (chfast) [inline README.md]: How about naming it `abort-on-internal-error`?

File: src/hera.cpp [hunk 2]
```diff
@@ -371,12 +372,18 @@ evmc_result hera_execute(
...
     ret.status_code = EVMC_INTERNAL_ERROR;
     HERA_DEBUG << "Unknown exception: " << e.what() << "\n";
+    if (hera->fuzzing)
+      abort();
   } catch (...) {
     ret.status_code = EVMC_INTERNAL_ERROR;
     HERA_DEBUG << "Totally unknown exception\n";
+    if (hera->fuzzing)
+      abort();
   }
```
Comment (chfast) [inline src/hera.cpp]:
```suggestion
if (ret.status_code == EVMC_INTERNAL_ERROR)
  abort();
```

File: src/hera.cpp [hunk 3]
```diff
@@ -371,12 +372,18 @@ evmc_result hera_execute(
   } catch (InternalErrorException const& e) {
     ret.status_code = EVMC_INTERNAL_ERROR;
     HERA_DEBUG << "InternalError: " << e.what() << "\n";
+    if (hera->fuzzing)
```
Comment (chfast) [inline src/hera.cpp]: You can replace these checks with single one suggested below.

File: src/hera.cpp (hunk 2 above)
Comment (axic) [reply src/hera.cpp]: True. Though probably I prefer this way because the branch is only triggered in case of an error.

File: README.md (hunk 1 above)
Comment (axic) [reply README.md]: Any shorter suggestion?

Comment (chfast) [reply README.md]: No.

Comment (axic) [reply README.md]: A suggestion: "errormode", with possible value "abort-on-internal-error".

**Comment by axic** [comment]
This may be obsolete now.

**Pull request closed without merging**